dev/tools/shared/node.py

```
@@ -122,10 +127,21 @@ def install_node(install_dir):
-    except Exception as e:
-        print(f"Failed to download Node.js: {e}")
+    # The nodejs.org CDN occasionally drops connections mid-download in CI;
+    # retrying a couple of times recovers most of those without masking a
+    # persistent outage (e.g. a bad version/URL, which fails every attempt).
+    last_error = None
+    for attempt in range(1, _DOWNLOAD_ATTEMPTS + 1):
+        try:
+            urllib.request.urlretrieve(url, tar_path)
+            last_error = None
+            break
+        except Exception as e:
```

Ruff (0.16.3)

[warning] Do not catch blind exception: `Exception`

(BLE001)